Assets/Scripts/Towers/Tower.cs

```diff
@@ -74,23 +103,54 @@ public class Tower : MonoBehaviour {
 
     private void UpdateTarget() {
         PurgeDestroyedEnemies();
+        PurgeOutOfRangeEnemies();
+
+        // If multitarget is at max but there are still other enemies in range
+        if (currentTargets.Count == towerType.targetCount && currentTargets.Count < enemiesInRange.Count) {
+            switch (towerType.targetingMethod) {
+                case TowerType.TargetingMethod.HighestPriority:
+                    FindAnyHigherPriorityEnemy();
+                    break;
+                case TowerType.TargetingMethod.LowestPriority:
+                    FindAnyLowerPriorityEnemy();
+                    break;
+                case TowerType.TargetingMethod.Closest:
+                    FindAnyCloserEnemy();
+                    break;
+            }
+        }
 
-        currentTarget = towerType.targetingMethod switch {
-            TowerType.TargetingMethod.HighestPriority => FindHighestPriorityEnemy(),
-            TowerType.TargetingMethod.LowestPriority => FindLowestPriorityEnemy(),
-            TowerType.TargetingMethod.Closest => FindClosestEnemy(),
-            _ => null
-        };
+        while (currentTargets.Count < towerType.targetCount && currentTargets.Count < enemiesInRange.Count) {
+            currentTargets.Add(new Target {
+                Enemy = towerType.targetingMethod switch {
+                    TowerType.TargetingMethod.HighestPriority => FindHighestPriorityEnemy(),
+                    TowerType.TargetingMethod.LowestPriority => FindLowestPriorityEnemy(),
+                    TowerType.TargetingMethod.Closest => FindClosestEnemy(),
+                    _ => throw new System.NotImplementedException()
+                }
+            });
+        }
+    }
+
+    private void FindAnyHigherPriorityEnemy() {
+        currentTargets.ForEach(target => {
```

Comment:
this is a thrice nested loop (although inner-most Any may terminate early). Would probably be more efficient just finding the highest prioritized targets in enemiesInRange without checking currentTargets